script.js

```diff
@@ -0,0 +1,621 @@
+// Global setup ====================================================
+// Information related to player -----------------------------------
+// points player start with
+const PLAYER_STARTING_POINTS = 100;
+// the player's total points
+let playerTotalPoints = PLAYER_STARTING_POINTS;
+// player's bid points
+let playerBidPoints = 0;
+// array to store player's hand cards
+let playerHand = []; // playerHand = ['']; // to use for testing only
+// player hand size
+const handSize = 5;
+// array to store player's cards to exchange
+let cardsToExchange = [];
+
+// shuffled deck
+let deck;
+
+// Elements and containers to display items in browser ------------
+// button to deal cards
+let dealButton;
+// button to exchange cards
+let exchangeOrHoldCardsButton;
+// container to display buttons
+let buttonsContainer;
+// container to display the player's cardContainers
+let playerHandContainer;
+// container to display output messages informing player about the state of the game
+let gameInfo;
+// element to display heading for player's total points
+let totalPointsHeadingEl;
+// element to display player's total points
+let totalPointsInfoEl;
+// container to display player's total points info and heading
+let totalPointsContainer;
+// element to display heading for player's bid points
+let bidPointsHeadingEl;
+// element to display player's bid points
+let bidPointsInfoEl;
+// input element to ask for points player wants to bid
+let bidPointsInputEl;
+// button to submit the points player wants to bid
+let bidPointsButton;
+// container to display player's bid points info and heading
+let bidPointsContainer;
+// container to display player's bid points input box and button
+let bidPointsInputContainer;
+// container that other containers in this file will append to
+const gameDisplayContainer = document.getElementById('game-display');
+
+// Variables used in calculating hand score ----------------------
+// to store cards based on similar ranks
+let rankedHand = [[]];
+// to find number of 4 of a kind in playerHand
+let numOf4OfAKind = 0;
+// to find number of 3 of a kind in playerHand
+let numOf3OfAKind = 0;
+// to find number of pairs in playerHand
+let numOfPairs = 0;
+// points of the player hand
+let handScore = 0;
+
+// For controlling button clickability ----------------------------
+// only allow player to submit bid points at start of the round
+let canSubmitBidPoints = true;
+// Only allow player to click on dealButton after submitting bid points and before exchanging cards
+let canDealStartingCards = false;
+// only allow player to click on card after having dealt cards
+let canClickCards = false;
+// only allow player to click on exchangeOrHoldCardsButton after having dealt cards
+let canExchangeOrHoldCards = false;
+
+// Helper functions ================================================
```

Comment:
Can start to be more specific- e.g. pretty much all functions now are "helper" functions. Can we use more specific groupings of functions instead of just "helper"?